internal/messaging/response_handler.go

```diff
@@ -169,7 +118,9 @@ func (h *evmResponseHandler) handleMintResponse(ctx context.Context, response *R
 	h.logger.Debugf("Token URI JSON: %s\n", jsonPlain)
 
 	if response.MintResponse.BuyableUntil == nil || response.MintResponse.BuyableUntil.Seconds == 0 {
-		response.MintResponse.BuyableUntil = timestamppb.New(time.Now().Add(h.buyableUntilDefault))
+		response.MintResponse.BuyableUntil = timestamppb.New(time.Now().Add(300 * time.Second))
+	} else if timestamppb.New(time.Now()).Seconds-response.MintResponse.BuyableUntil.Seconds < 70 {
```

Comment:
Isn't this comparison need to be the other way around? You are subtracting the buyable until value from the current timestamp. This would result in a negative number.

Reply:
And if the `BuyableUntil.Seconds` is already at this stage less than `time.Now` ho` should it be handeled?

Reply:
if buyable_until == nil/zero --> set it to set it to current_timestamp + 300s`
if buyable_until < current_timestamp --> Error
if buyable_until < current_timestamp + 70s --> set it to current_timestamp + 70s
if buyable_until > current_timestamp + 600s --> set it to current_timestamp + 600s